### PR TITLE
chore: update polkadot dependencies

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes KILTProtocol/ticket#1283

Updates dependencies from the `@polkadot` scope to version ^4 which was required for the next v0.9.0 Polkadot/Kusama chain upgrades where parachains are used according to their [release notes](https://github.com/polkadot-js/api/releases/tag/v4.9.1).
This required updating typescript to version ^4 as well, and bumps node requirements to >= 14. Github actions are now run against node v14, additionally running unit tests against node v16 (soon to be LTS).  
With the update we also saw the ErrorHandler implementation breaking, which required some changes there. 

## How to test:
Integration tests and unit tests should cover a lot of ground; feel free to run additional tests if you can come up with something.

## Checklist:

- [x] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/chain-helpers/src/errorhandling/ErrorHandler.spec.ts [hunk 1]
```diff
@@ -24,20 +23,17 @@ describe('ErrorHandler', () => {
         },
       },
     }
-    const submittableResult: ISubmittableResult = {
-      // @ts-ignore
-      events: [evtRecord],
-    }
+    const submittableResult = {
+      events: ([evtRecord] as unknown) as EventRecord[],
```
Comment (tjwelde) [inline packages/chain-helpers/src/errorhandling/ErrorHandler.spec.ts:27]: a lot of type assertions here. Is there a way to make it work without enforcing the types?

File: packages/core/src/requestforattestation/RequestForAttestation.ts [hunk 2]
```diff
@@ -206,7 +206,7 @@ export default class RequestForAttestation implements IRequestForAttestation {
    * ```
    */
   public removeClaimOwner(): void {
-    delete this.claim.owner
+    delete (this.claim as Partial<IClaim>).owner
```
Comment (tjwelde) [inline packages/core/src/requestforattestation/RequestForAttestation.ts:209]: kinda hacky (though I know that the deletion doesn't fit with the defined type of `this.claim`). What about the `@ts` comments you did in the test files. Is there one, where we can say: "just let it be"?

File: jest.config.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
 module.exports = {
-  preset: 'ts-jest',
+  preset: 'ts-jest/presets/js-with-ts',
```
Comment (tjwelde) [inline jest.config.js:2]: why is this necessary?

**Review by tjwelde — COMMENTED** [review]
Looks good overall, just some questions and nitpicks

File: packages/chain-helpers/src/errorhandling/ErrorHandler.spec.ts (hunk 1 above)
Comment (rflechtner) [reply packages/chain-helpers/src/errorhandling/ErrorHandler.spec.ts:27]: `@ts-ignore` as we had it before :D

Comment (rflechtner) [reply packages/chain-helpers/src/errorhandling/ErrorHandler.spec.ts:27]: this is testing, right, so we're using a mocked EventRecord. Could potentially construct the real thing using the TypeRegistry, but is it worth the trouble?

File: packages/core/src/requestforattestation/RequestForAttestation.ts (hunk 2 above)
Comment (rflechtner) [reply packages/core/src/requestforattestation/RequestForAttestation.ts:209]: it was `@ts-ignore` before; if you like that better, I can revert. We have a linter rule against its usage though, so we need to escape that as well.

File: jest.config.js (hunk 3 above)
Comment (rflechtner) [reply jest.config.js:2]: this transpilesjs files, including polkadot dependencies, to CommonJS syntax. This is because jest can't really handle esm yet

File: packages/core/src/requestforattestation/RequestForAttestation.ts (hunk 2 above)
Comment (tjwelde) [reply packages/core/src/requestforattestation/RequestForAttestation.ts:209]: I would rather want to have a warning for it.
So if the linting is configured to show it as an error, let's reconfigure it to list it as warning.
Because it is actually not the correct way to do it and we will need the reminder.

File: packages/chain-helpers/src/errorhandling/ErrorHandler.spec.ts (hunk 1 above)
Comment (tjwelde) [reply packages/chain-helpers/src/errorhandling/ErrorHandler.spec.ts:27]: oh yeah, sure. You are right. Not worth the trouble

**Review by tjwelde — APPROVED** [review]
LGTM
